src/runner.ts

```diff
@@ -0,0 +1,578 @@
+  private _repos: string[];
+  private _skipTests: string[];
+  private _tests: string[];
+  private _testRateLimiter: Bottleneck;
+  private _verbose: boolean;
+  private _wctFlags: string;
+  private _workspace: Workspace;
+
+  // TODO(usergenic): This constructor is getting long.  Break up some of
+  // these stanzas into supporting methods.
+  constructor(options: RunnerOptions) {
+    this._excludeRepos = options.excludeRepos || [];
+    this._fresh = !!options.fresh;
+    // TODO(usergenic): Pass an option to gitUtil.connectToGitHub for the
+    // rate limiter it uses.
+    this._github = new git.GitHubConnection(options.githubToken);
+    this._repos = options.repos || [];
+    this._skipTests = options.skipTests || [];
+    this._tests = options.tests || [];
+    this._verbose = !!options.verbose;
+    this._wctFlags = options.wctFlags ? options.wctFlags.join(' ') : '';
+    this._workspace = {
+      dir: (options.workspaceDir || './tattoo_workspace'),
+      repos: new Map()
+    };
+
+    // TODO(usergenic): Rate limit should be an option.
+    this._testRateLimiter = new Bottleneck(1, 100);
+
+    if (this._verbose) {
+      console.log('Tattoo Runner configuration:');
+      console.log({
+        excludeRepos: this._excludeRepos,
+        fresh: this._fresh,
+        repos: this._repos,
+        skipTests: this._skipTests,
+        tests: this._tests,
+        wctFlags: this._wctFlags,
+        workspaceDir: this._workspace.dir
+      });
+    }
+  }
+
+  /**
+ * Analyzes all of the HTML in 'repos/*' with hydrolysis.
+ *
+ * @returns a promise of the hydrolysis.Analyzer with all of the info loaded.
+ */
+  async _analyzeRepos() {
+    const dirs = fs.readdirSync(this._workspace.dir);
+    const htmlFiles: string[] = [];
+
+    for (const dir of dirs) {
+      for (const fn of fs.readdirSync(path.join('repos', dir))) {
+        if (/index\.html|dependencies\.html/.test(fn) ||
+            !fn.endsWith('.html')) {
+          continue;
+        }
+        // We want to ignore files with 'demo' in them, unless the element's
+        // directory has the word 'demo' in it, in which case that's
+        // the whole point of the element.
+        if (!/\bdemo\b/.test(dir) && /demo/.test(fn)) {
+          continue;
+        }
+        htmlFiles.push(path.join('repos', dir, fn));
+      }
+    }
+
+    function filter(repo: string) {
+      return !util.existsSync(repo);
+    }
+
+    // This code is conceptually simple, it's only complex due to ordering
+    // and the progress bar. Basically we call analyzer.metadataTree on each
+    // html file in sequence, then finally call analyzer.annotate() and return.
+    const analyzer = await hydrolysis.Analyzer.analyze(
+        path.join(this._workspace.dir, 'polymer', 'polymer.html'), {filter});
+
+    const progressBar =
+        util.standardProgressBar('Analyzing...', htmlFiles.length + 1);
+
+    for (const htmlFile of htmlFiles) {
+      await analyzer.metadataTree(htmlFile);
```

Comment:
Moot.  Function removed in 9a88a6e0804a7b3da121d9f8944060c92af48f80 after it was discovered to be not being called anywhere.